posts/discriminative-vs-generative/index.qmd

```diff
@@ -0,0 +1,595 @@
+$$
+P_d \left( D, \theta, \phi \right) =: P^1 \left( y, \theta \vert x \right) P^2 \left( x, \phi \right),
+$$
+then we see that Bayes' update rule simplifies, namely
+$$
+\begin{align*}
+P \left( \theta, \phi \vert D \right)   &= \frac{P_d \left( D, \theta, \phi \right)}{\sum_{\theta, \phi} P_d \left( D, \theta, \phi \right)} \\
+                                        &= \frac{P^1 \left( y, \theta \vert x \right) P^2 \left( x, \phi \right)}{\sum_{\theta , \phi} P^1 \left( y, \theta \vert x \right) P^2 \left( x, \phi \right)} \\
+                                        &= \frac{P^1 \left( y, \theta \vert x \right)}{\sum_\theta P^1 \left( y, \theta \vert x \right) } \frac{P^2 \left( x, \phi \right)}{\sum_\phi P^2 \left( x, \phi \right)} \\ 
+                                        &=: P \left( \theta \vert D \right) P \left( \phi \vert x \right).
+\end{align*}
+$$
+
+```{python}
+#| fig-cap: Comparison of parameter space considered by models. The generative model only considers the hyperplane $\theta = \phi.$
+#| column: margin
+
+x = [0,1]
+y = x
+
+plt.figure(figsize=(1.2,1.2))
+
+line, = plt.plot(x, y)
+line.set_color('black')
+
+ax = plt.gca()
+
+ax.annotate(
+    'generative \n model', 
+    xy=(0.5, 0.5),  
+    xycoords='data',
+    xytext=(1.2, 0.5),
+    arrowprops=dict(width=1, headwidth=5, facecolor='black', shrink=0.05),
+    horizontalalignment='left', verticalalignment='center',
+            )
+
+ax.annotate(
+    'discriminative \n model', 
+    xy=(0.6, 0.15),  
+    xycoords='data',
+    xytext=(1.2, 0.15),
+    arrowprops=dict(width=1, headwidth=5, facecolor='black', shrink=0.05),
+    horizontalalignment='left', verticalalignment='center',
+            )
+
+ax.set_aspect('equal', adjustable='box')
+ax.xaxis.set_tick_params(labelbottom=False) 
+ax.yaxis.set_tick_params(labelleft=False) 
+ax.set_xticks([]) 
+ax.set_yticks([])
+plt.xlabel(r"$\theta$")
+plt.ylabel(r"$\phi$")
+plt.title("Parameter space")
+
+# plt.savefig('params.png')
+plt.show()
+```
+
+Thus $\hat \theta$ (or equivalently $P \left( \theta \vert D \right)$) is unaffected by the estimation of $\hat \phi$, and is the same as what we would have obtained by performing inference on the original, non-extended discriminative model.
+
+We can now see that the fundamental difference between the two models is down to the discriminative one considering a larger parameter space without the constraint $\theta = \phi$.
+This reduces the (statistical) bias, but introduces variance.
+
+We have just shown that the generative model is more biased, but this is a more fundamental difference than what is usually discussed. Indeed, the joint distribution considered by the generative model is often hard to work with in practice and therefore further simplifying assumptions are often necessary, or preferable, in order to make inference tractable. This can lead to further bias of the generative model, which we will consider next.
```

Comment:
I would reformulate this. I'd drop the "but this is a more" part and write this more directly again. Maybe something like "We therefore see that generative models are more biased in the sense that they consider a more restricted parameter space. This view presents a more fundamental distinction between the two types of models than is usually discussed" or something similar?